cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -591,6 +592,13 @@ func allPodsAreNew(pods []*apiv1.Pod, currentTime time.Time) bool {
 	return found && oldest.Add(unschedulablePodWithGpuTimeBuffer).After(currentTime)
 }
 
+func buildNodeForNodeTemplate(nodeTemplate *schedulernodeinfo.NodeInfo, index int) *apiv1.Node {
```

Comment:
Looks good. 
@vivekbagade is doing some testing on this (thanks!).
I will LGTM when we are done with that.